app/controllers/certificates_controller.rb

```diff
@@ -17,8 +17,12 @@ def edit
 
   def destroy
     @certificate = Certificate.find(params[:id])
-    @certificate.destroy!
-    redirect_to(certificates_path, notice: "Successfully removed Certificate: #{@certificate.name}")
+    if @certificate.has_child?
+      redirect_to(certificates_path, alert: "Cannot remove a certificate with issued certificates. Please remove the issued certificates first.")
```

Comment:
Is this to do with intermediates or roots, maybe more clearer wording like, "unable to remove intermediate certificate without removing the client first" or something like that.

Reply:
it depends, it can be any certificate that has direct children, so either a root or any intermediate. In x509 speak, people seem to not use 'child' certificate so much and 'issuing' seems to be the word? Happy to change it to something else.

Reply:
Ok, I understand, ya that's tricky if there isn't standard thermology,